File: Q(10-05-2023).cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int n;
    cin >> n;
    
    vector<char> initial(n), rotated(n);
    for (int i = 0; i < n; i++) {
        cin >> initial[i];
    }
    for (int i = 0; i < n; i++) {
        cin >> rotated[i];
    }
    
    int index = -1;
    for (int i = 0; i < n; i++) {
        if (initial[0] == rotated[i]) {
            index = i;
            break;
        }
    }
    
    bool shuffled = false;
    for (int i = 0; i < n; i++) {
        if (initial[i] != rotated[(i + index) % n]) {
            shuffled = true;
            break;
        }
    }
    
    if (shuffled) {
        cout << "Shuffled" << endl;
    } else {
        cout << "Not Shuffled" << endl;
    }
    
    return 0;
}

File: Q(10-05-2023).java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int n = scanner.nextInt();
        scanner.nextLine();

        String[] initial = new String[n];
        String[] rotated = new String[n];

        for (int i = 0; i < n; i++) {
            initial[i] = scanner.nextLine().trim();
        }

        for (int i = 0; i < n; i++) {
            rotated[i] = scanner.nextLine().trim();
        }

        int index = -1;

        for (int i = 0; i < n; i++) {
            if (rotated[i].equals(initial[0])) {
                index = i;
                break;
            }
        }

        for (int i = 0; i < n; i++) {
            if (!rotated[(index + i) % n].equals(initial[i])) {
                System.out.println("Shuffled");
                return;
            }
        }

        System.out.println("Not Shuffled");
    }
}
